Libs/Image/Image.cpp

```diff
@@ -391,4 +393,82 @@ bool Image::pad(int padding, PixelType value)
 #endif
   return true;
 }
+
+bool Image::warp(std::string &source_file, std::string &target_file, int pointFactor)
+{ 
+  if (!this->image)
+  {
+    std::cerr << "No image loaded, so returning false." << std::endl;
+    return false;
+  }
+  using ResamplerType = itk::ResampleImageFilter< ImageType, ImageType>;
+  typedef   itk::LinearInterpolateImageFunction< ImageType, double > InterpolatorType;
+  // first need to read the source and target container
+  PointSetType::Pointer sourceLandMarks = PointSetType::New();
+  PointSetType::Pointer targetLandMarks = PointSetType::New();
+  PointType p1;     
+  PointType p2;
+  PointSetType::PointsContainer::Pointer sourceLandMarkContainer = sourceLandMarks->GetPoints();
+  PointSetType::PointsContainer::Pointer targetLandMarkContainer = targetLandMarks->GetPoints();
+  PointIdType id = itk::NumericTraits< PointIdType >::Zero;
+  std::ifstream insourcefile;
+  std::ifstream intargetfile;
+  insourcefile.open( source_file.c_str());
+  intargetfile.open( target_file.c_str());
+  int count = 0;
+
+  while (!insourcefile.eof() && !intargetfile.eof()){
+      insourcefile >>  p1[0] >> p1[1] >> p1[2];
+      intargetfile >> p2[0] >> p2[1] >>p2[2];
+      if(count % pointFactor == 0){
+        // reverse warp thats why reverse order of p1 p2
+      targetLandMarkContainer->InsertElement( id, p1 );
+      sourceLandMarkContainer->InsertElement( id, p2 );
+      id++;
+      }
+      count++;
+  }
+  insourcefile.close();
+  intargetfile.close();
+
+  // now perform the warping
+  TransformType::Pointer tps = TransformType::New();
+  tps->SetSourceLandmarks(sourceLandMarks);
+  tps->SetTargetLandmarks(targetLandMarks);
+  tps->ComputeWMatrix();
+  // Set the resampler params
+  ResamplerType::Pointer resampler = ResamplerType::New();
+  InterpolatorType::Pointer interpolator = InterpolatorType::New();
+  resampler->SetInterpolator( interpolator );
+  ImageType::SpacingType spacing = image->GetSpacing();
+  ImageType::PointType   origin  = image->GetOrigin();
+  ImageType::DirectionType direction  = image->GetDirection();
+  ImageType::RegionType region = image->GetBufferedRegion();
```

Comment:
Image::applyTransform by default applies it to the whole image. GetBufferedRegion is unnecessary and only affects threading, which is also used by default if possible for applying the filter. For the way this function is written, no threading would be possible anyway, but for our Image::applyTransform maybe there's a chance (I'm still skeptical).